update.ps1

```diff
@@ -160,208 +107,239 @@ function Resolve-ZivverError {
-            $responseUser = $null
+            $correlatedAccount = $null
         }
         else {
             throw
         }
     }
+    $outputPreviousData = $correlatedAccount.PsObject.Copy()
+    Write-Verbose "Queried Ziver account where [id] = [$($actionContext.References.Account)]. Result: $($correlatedAccount | ConvertTo-Json)"
+    #endregion Get Zivver account
+
+    #region Calulate action
+    $actionMessage = "calculating action"
+    if (($correlatedAccount | Measure-Object).count -eq 1) {
+        $actionMessage = "comparing current account to mapped properties"
+
+        # Change mapping here
+
+        # Define your mapping here for correlatedaccount data to compare
+        $correlatedReferenceObject = [PSCustomObject]@{
+            fullname = $correlatedAccount.name.formatted
+            division = $correlatedAccount.'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User'.division
+            userName = $correlatedAccount.userName
+        }
 
-    if ($responseUser.Length -lt 1) {
-        throw "Zivver account for: [$($p.DisplayName)] not found. Possibly deleted"
-    }
+        # Define your mapping here for fieldmapping data to compare
+        $accountDifferenceObject = [PSCustomObject]@{
+            fullname = $account.name.formatted
+            division = $account.'urn:ietf:params:scim:schemas:extension:enterprise:2.0:User'.division
+            userName = $account.userName
+        }
 
-    Write-Verbose "Verify if Zivver account for [$($p.DisplayName)] must be updated"
-    $splatCompareProperties = @{
-        ReferenceObject   = @($responseUser)
-        DifferenceObject  = @($account)
-        ExcludeProperties = @("delegates", "id", "meta", "active", "urn:ietf:params:scim:schemas:extension:enterprise:2.0:User", "urn:ietf:params:scim:schemas:zivver:0.1:User", "schemas") # Properties not managed by HelloID, are excluded from the comparison.
+        $splatCompareProperties = @{
+            ReferenceObject  = @($correlatedReferenceObject.PSObject.Properties)
+            DifferenceObject = @($accountDifferenceObject.PSObject.Properties)
+        }  
+        $accountPropertiesChanged = Compare-Object @splatCompareProperties -PassThru
+        $accountOldProperties = $accountPropertiesChanged | Where-Object { $_.SideIndicator -eq "<=" }
+        $accountNewProperties = $accountPropertiesChanged | Where-Object { $_.SideIndicator -eq "=>" }
+
+        if ($accountNewProperties) {
+            $actionAccount = "Update"
+            Write-Information "Account property(s) required to update: $($accountNewProperties.Name -join ', ')"
+        }
+        else {
+            $actionAccount = "NoChanges"
+        }    
     }
-    $propertiesChanged = Compare-ZivverAccountObject @splatCompareProperties
-
-    if ($propertiesChanged) {
-        # Create the JSON body for the properties to update
+    elseif (($correlatedAccount | Measure-Object).count -gt 1) {
```

Comment:
deze zou je, als je kijkt naar de aanroep van de GET call `/Users/{id}` niet verwachten. In de create is deze wel weer logisch omdat we daar werken met een filter. (Die een array van users teruggeeft).

Reply:
In het kader van FFP is het wel beter om "MultipleFound" overal weg te slopen behalve in de create, zal ik nog aanpassen